src/plugin.ts

```diff
@@ -5,20 +5,25 @@ export default {
   name: "vite:relay",
   transform(src, id) {
     let code = src;
+
     if (/.(t|j)sx?/.test(id) && src.includes("graphql`")) {
       const out = transformSync(src, {
         plugins: [
           [
-            require.resolve("babel-plugin-relay"),
+            "babel-plugin-relay",
             {
-              eagerESModules: true,
+              eagerEsModules: true,
             },
           ],
         ],
         code: true,
+        filename: id,
```

Comment:
This forwards the filename to `transformSync`, so `babel-plugin-relay` can correctly compute import paths, when using `artifactDirectory`.